File: ocUseFlutter/ProjectObjectiveC/ProjectObjectiveC/RootVC.m
//
//  RootVC.m
//  ProjectObjectiveC
//
//  Created by lining02 on 2024/3/9.
//

#import "RootVC.h"
#import <Flutter/Flutter.h>
@interface RootVC ()
@property(nonatomic,strong) UIButton * jumpBtn;
@property(nonatomic,strong) FlutterViewController * flutterVC;
@property(nonatomic,strong) FlutterMethodChannel * dismissChannel;

@end

@implementation RootVC

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor redColor];
    [self.view addSubview:self.jumpBtn];
}

- (UIButton *)jumpBtn {
    if (!_jumpBtn) {
        _jumpBtn = [UIButton buttonWithType:UIButtonTypeSystem];
        _jumpBtn.frame = CGRectMake(0, 200, 200, 60);
        [_jumpBtn setTitle:@"跳转Flutter页面" forState:UIControlStateNormal];
        [_jumpBtn setTitle:@"跳转Flutter页面" forState:UIControlStateSelected];
        [_jumpBtn setBackgroundColor:[UIColor orangeColor]];
        [_jumpBtn addTarget:self action:@selector(clickjumpBtn) forControlEvents:UIControlEventTouchUpInside];
    }
    return _jumpBtn;
}

- (FlutterViewController *)flutterVC {
    if (!_flutterVC) {
        _flutterVC = [[FlutterViewController alloc]init];
        _flutterVC.modalPresentationStyle = UIModalPresentationFullScreen;
        
        _dismissChannel = [FlutterMethodChannel methodChannelWithName:@"FlutterHomePage.channel" binaryMessenger:_flutterVC.binaryMessenger];
        
        __weak typeof(self) weakSelf = self;
        [_dismissChannel setMethodCallHandler:^(FlutterMethodCall* call, FlutterResult result) {
            if ([@"dismiss" isEqualToString:call.method]) {
                
                [weakSelf dismissViewControllerAnimated:true completion:^{
                    
                }];
                
            } else {
                result(FlutterMethodNotImplemented);
            }
        }];
        
    }
    return _flutterVC;
}


- (void)clickjumpBtn {
    
    [self presentViewController:self.flutterVC animated:YES completion:nil];
}


@end

File: SwiftUseSwiftUI/ProjectSwift/RootVC.swift
//
//  RootVC.swift
//  ProjectSwift
//
//  Created by pandakingli on 2024/3/29.
//

import UIKit
import SwiftUI


class RootVC: UIViewController {
    
    var jumpbtn: UIButton = UIButton(type: UIButton.ButtonType.custom)
    var uiMyVC : UIViewController?
    @objc func clickjumpbtn() {
        print("clickjumpbtn")
        if #available(iOS 13.0, *) {
            let uiV = mySViewOne {
                if self.uiMyVC != nil {
                    self.uiMyVC!.dismiss(animated: true)
                    self.uiMyVC = nil
                }
            }
            let vc = UIHostingController(rootView: uiV)
            vc.modalPresentationStyle = .fullScreen
            uiMyVC = vc
            
            self.present(vc, animated: true)
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.view.backgroundColor = UIColor.blue

        jumpbtn.frame = CGRect(x: 100, y: 200, width: 100, height: 60)
        jumpbtn.backgroundColor = UIColor.orange
        jumpbtn.setTitle("跳转", for: UIControl.State.normal)
        jumpbtn.addTarget(self, action: #selector(clickjumpbtn), for: UIControl.Event.touchUpInside)
        self.view.addSubview(jumpbtn)
        
    }

}
